tensorboard/webapp/core/store/core_selectors.ts

```diff
@@ -35,13 +36,27 @@ export const getPluginsListLoaded = createSelector(
   (state: CoreState): PluginsListLoadState => state.pluginsListLoaded
 );
 
-// TODO(tensorboard-team): AppLastLoaded is currently derived from plugins listing loaded
-// state which should be disentangled. Fix this by having a separate state for remembering
-// when the application data was last loaded.
-export const getAppLastLoadedTimeInMs = createSelector(
-  getPluginsListLoaded,
-  (loadedState: PluginsListLoadState): number | null =>
-    loadedState.lastLoadedTimeInMs
+export const getPolymerRunsLoadState = createSelector(
+  selectCoreState,
+  (state: CoreState): LoadState => state.polymerRunsLoadState
+);
+
+export const getCoreDataLoadedState = createSelector(
+  selectCoreState,
+  (state: CoreState): DataLoadState => {
+    return state.coreDataLoadState.state;
+  }
+);
+/**
+ * Returns last _successful_ loaded time of the applicational state where an
+ * applicational state is defined as combinations of plugins listing, runs, and
+ * environment.
```

Comment:
Is it worth clarifying that `environment` does not, in reality, impact CoreDataLoadedState? IIRC you previously modified the `TODO(tensorboard-team)` to clarify that.

Reply:
Added the TODO back. `environment` does impact `CoreDataLoadedState`. If it fails to load, the effects will dispatch `fetchPluginsListingFailed` and it ultimately changes the `CoreDataLoadedState` to failed.